doc/tutorials/introduction/transition_guide/transition_guide.markdown

```diff
@@ -257,3 +257,49 @@ _cuda_ module has been split into several smaller pieces:
 Documentation format {#tutorial_transition_docs}
 --------------------
 Documentation has been converted to Doxygen format. You can find updated documentation writing guide in _Tutorials_ section of _OpenCV_ reference documentation (@ref tutorial_documentation).
+
+Support both versions {#tutorial_transition_both}
+---------------------
+In some cases it is possible to support both versions of OpenCV.
+
+### Source code
+
+To check library major version one of the following methods can be used:
+- if __CV_VERSION_EPOCH__ is defined - you are using 2.4 branch, otherwise - 3.x
+- __CV_MAJOR_VERSION__ is defined as `2` or `3` for corresponding version
+
+One of `opencv2/core/version.hpp` or `opencv2/core/core.hpp` files should be included to use these definitions.
+
+Examples:
+@code{.cpp}
+#ifdef CV_VERSION_EPOCH
+// do opencv 2 code
+#else
+// do opencv 3 code
+#endif
+@endcode
+or
+@code{.cpp}
+#if CV_MAJOR_VERSION == 2
```

Comment:
I believe, this method should be presented first, as the main one. The method with CV_VERSION_EPOCH is not obvious and may be confusing for someone who reads code and does not know about this trick. Thus it should either be not mentioned at all, or you can describe it as alternative hacky solution